### ISSUE TITLE
[BUG] (Problems with marlin2.0.x & bugfix2.0 on mks sbase v1.3 and external drivers)

### ISSUE THREAD

**Issue by rouhit110** [MarlinFirmware/Marlin#23610]
### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

Hi, I can not establish a stable connection between the controller and the drivers. My drivers are leadshine DM870 & DM556
I tried a lot with Marlin 2.0.9.3 but it didn't work
I made the necessary settings in all the tabs according to the experts' instructions and explanations.
Thermistors and heaters work but the motors do not move.
With Marlin Bug Fix 2.0, I only managed to move the steppers once, but after restarting the controller, they failed.

Configuration.h "bugfix2.0"
```cpp
#define X_DRIVER_TYPE  TB6600
#define Y_DRIVER_TYPE  TB6600
#define Z_DRIVER_TYPE  TB6600
#define E0_DRIVER_TYPE TB6600

#define X_ENABLE_ON 1
#define Y_ENABLE_ON 1
#define Z_ENABLE_ON 1
#define E_ENABLE_ON 1
```
Configuration_adv.h "bugfix2.0"
```cpp
#define INVERT_X_STEP_PIN false
#define INVERT_Y_STEP_PIN false
#define INVERT_Z_STEP_PIN false
#define INVERT_E_STEP_PIN false
```
This's wiring my controller to the drivers
from Drivers to Controller
EN-   to   EN
EN+  to  +5V
DIR-  to   DIR
DIR+ to  +5V
PUL-  to  STEP
PUL+ to +5V
With these settings, the steppers worked for once and did not work after resetting the controller

Needless to say, I did not have stepper movement problems in other firmware, but I had problems controlling the heaters.

Thank you in advance for your help and guidance friends

### Version of Marlin Firmware

marlin 2.0 & bugfix2.0

### Bed Leveling

No Bed Leveling

### Your Slicer

Simplify3D

### Host Software

SD Card (headless)

**Comment by ellensp** [comment]
```
This's wiring my controller to the drivers
from Drivers to Controller
EN- to EN
EN+ to +5V

```
So when Enable is high, EN- is high and EN+ is always high. So the driver is disabled

For your wiring, all these need to be 0 not 1
#define X_ENABLE_ON 1
#define Y_ENABLE_ON 1
#define Z_ENABLE_ON 1
#define E_ENABLE_ON 1


Or re-wire with EN- tied to gnd and EN+ tied to the enable pin so that  ENABLE_ON 1 works as desired

**Comment by ellensp** [comment]
Also mks sbase v1.3 is a 3.3v device...  not 5v..   
So your getting weird logic levels. 
so En- is 3.3v and en+ is 5v   is  that a logical 1 or 0? I suspect it indeterminate.

**Comment by rouhit110** [comment]
> Also mks sbase v1.3 is a 3.3v device... not 5v.. So your getting weird logic levels. so En- is 3.3v and en+ is 5v is that a logical 1 or 0? I suspect it indeterminate.

By changing the jumper to 3.3 volts, the steppers started working. thank you so much
But does this not cause a problem for the BLTouch and MKS PWC sensors?

**Comment by ellensp** [comment]
on lpc1768 most pins are 5v tolerant (only in digital mode)

**Comment by rouhit110** [comment]
> on lpc1768 most pins are 5v tolerant (only in digital mode)

Thank you for your answering dear ellensp.

**Comment by rouhit110** [comment]
> on lpc1768 most pins are 5v tolerant (only in digital mode)

Unfortunately, all endstops are down now. In Marlin Bug Fix 2.0, I enabled pullup and pulldown settings one after the other, but it didn't work. 
My endstops are optical and three-wire
I also connected the voltage wire from 3.3 to 5 volts, but it was not effective
Do you come up with something special?

**Comment by ellensp** [comment]
These are general support questions. not bug reports

This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- MarlinFW.org [Marlin Documentation](http://marlinfw.org)
- RepRap.org [Marlin Forum](http://forums.reprap.org/list.php?415)
- Facebook Group ["Marlin Firmware"](https://www.facebook.com/groups/1049718498464482/)
- Facebook Group ["Marlin Firmware for 3D Printers"](https://www.facebook.com/groups/3Dtechtalk/)
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- 3D Printing Discord server. Join link: https://discord.gg/qNnSzsug
- Marlin specific Discord server. Join link: https://discord.gg/n5NJ59y

After seeking help from the community, if the consensus points to a bug in Marlin, then you should post a [bug report](https://github.com/MarlinFirmware/Marlin/issues/new/choose).
